source/grid/grid_generator.cc

```diff
@@ -1910,6 +1910,259 @@ namespace GridGenerator
+        double length = std::numeric_limits<double>::max();
+        for (const auto cell : tria.active_cell_iterators())
+          for (unsigned int n = 0; n < GeometryInfo<2>::lines_per_cell; ++n)
+            length = std::min(
+              length,
+              (cell->line(n)->vertex(0) - cell->line(n)->vertex(1)).norm());
+        return length;
+      };
+
+      // start by setting up the cylinder triangulation
+      Triangulation<2> cylinder_tria;
+      GridGenerator::hyper_cube_with_cylindrical_hole(cylinder_tria,
+                                                      inner_radius,
+                                                      outer_radius,
+                                                      L,
+                                                      /*repetitions*/ 1,
+                                                      colorize);
+
+      // Mark the cylinder material ids as 2 to distinguish them from the bulk
+      // cells (these are read again about 40 lines down)
+      for (const auto &cell : cylinder_tria.active_cell_iterators())
+        cell->set_material_id(2);
+
+      // hyper_cube_with_cylindrical_hole will have 2 cells along
+      // each face, so he element size is outer_radius
+
+      auto add_sizes = [](std::vector<double> &step_sizes,
+                          const double         padding,
+                          const double         h) -> void {
+        // use std::round instead of std::ceil to improve aspect ratio
+        // in case padding is only slightly larger than h.
+        const unsigned int num = std::round(padding / h);
+        for (unsigned int i = 0; i < num; ++i)
+          step_sizes.push_back(padding / num);
+      };
+
+      std::vector<std::vector<double>> step_sizes(2);
+      // x-coord
+      // left:
+      add_sizes(step_sizes[0], pad_left, outer_radius);
+      // center
+      step_sizes[0].push_back(outer_radius);
+      step_sizes[0].push_back(outer_radius);
+      // right
+      add_sizes(step_sizes[0], pad_right, outer_radius);
+      // y-coord
+      //   bottom
+      add_sizes(step_sizes[1], pad_bottom, outer_radius);
+      //   center
+      step_sizes[1].push_back(outer_radius);
+      step_sizes[1].push_back(outer_radius);
+      //   top
+      add_sizes(step_sizes[1], pad_top, outer_radius);
+
+      // now create bulk
+      Triangulation<2> bulk_tria;
+      const Point<2>   bl(-outer_radius - pad_left, -outer_radius - pad_bottom);
+      const Point<2>   tr(outer_radius + pad_right, outer_radius + pad_top);
+      GridGenerator::subdivided_hyper_rectangle(
+        bulk_tria, step_sizes, bl, tr, colorize);
+
+      // now remove cells reserved from the cylindrical hole
+      std::set<Triangulation<2>::active_cell_iterator> cells_to_remove;
+      for (const auto &cell : bulk_tria.active_cell_iterators())
+        {
+          if ((cell->center() - center).norm() < outer_radius)
```

Comment:
I'm concerned that this check won't work if you have a large outer radius and a very small amount of padding (e.g. `padding_top = outer_radius/1000` - this is a stupid example, but conceptually legitimate). Can you please check this out? If I'm correct then I think you may need to check the actual bounding box rather than use a radius-based condition.